static/app/bootstrap/initializeLocale.tsx

```diff
@@ -61,7 +62,8 @@ export async function initializeLocale(config: Config) {
   const queryStringLang = Array.isArray(queryString.lang)
     ? queryString.lang[0]
     : queryString.lang;
-  const languageCode = queryStringLang || config.languageCode || 'en';
+  const languageCode =
+    queryStringLang || config.user?.options?.language || config.languageCode || 'en';
```

Comment:
I think an argument can be mode to just ignore the system's language code and only use options.